test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/DefaultHtmlGeneratorTest.cs

```diff
@@ -190,6 +191,64 @@ public void GenerateTextArea_WithNullExpression_WithNameAttribute_DoesNotThrow()
             Assert.Equal(expected, attribute.Value);
         }
 
+        [Fact]
+        public void GenerateTextArea_WithMaxLengthValidationAttribute_AddsMaxLengthAttribute()
```

Comment:
Need similar tests for `AddTextBox(...)`, `AddPassword(...)` and perhaps `AddTextBox(...)` when `htmlAttributes` contains e.g. `type="search"`.